src/gui/sidebar/previews/layer/SidebarLayersContextMenu.cpp

```diff
@@ -0,0 +1,52 @@
+#include "gui/sidebar/previews/layer/SidebarLayersContextMenu.h"
+
+SidebarLayersContextMenu::SidebarLayersContextMenu(GladeGui* gui, SidebarToolbar* toolbar):
+        SidebarBaseContextMenu(gui->get("sidebarPreviewLayersContextMenu")) {
+
+    // Connect the context menu actions
+    const std::map<std::string, SidebarActions> ctxMenuActions = {
+            {"sidebarPreviewLayerMoveUp", SIDEBAR_ACTION_MOVE_UP},
+            {"sidebarPreviewLayerMoveDown", SIDEBAR_ACTION_MOVE_DOWN},
+            {"sidebarPreviewMergeDown", SIDEBAR_ACTION_MERGE_DOWN},
+            {"sidebarPreviewLayerDuplicate", SIDEBAR_ACTION_COPY},
+            {"sidebarPreviewLayerDelete", SIDEBAR_ACTION_DELETE},
+            // TODO(ja-he): add SIDEBAR_ACTION_NEW_BEFORE/AFTER if/when these are defined for layers
+    };
+
+
+    for (const auto& [name, action]: ctxMenuActions) {
+
+        GtkWidget* const entry = gui->get(name);
+        g_assert(entry != nullptr);
+
+        // Create callbacks, store the entry
+        using Data = ContextMenuData;
+        auto userdata = std::make_unique<Data>(Data{toolbar, action});
+
+        const auto callback =
+                G_CALLBACK(+[](GtkMenuItem* item, Data* data) { data->toolbar->runAction(data->actions); });
+        const gulong signalId = g_signal_connect(entry, "activate", callback, userdata.get());
+        g_object_ref(entry);
+        this->contextMenuSignals.emplace_back(entry, signalId, std::move(userdata));
+
+        if (name == "sidebarPreviewLayerMoveUp") {
```

Comment:
Please do not duplicate strings in the classes. This can lead to really awful bugs caused by typos. Instead, predefine the strings in a shared context and reuse them through the class:

```
constexpr char* constString = "something";
```

Reply:
Fair point, again the code structure mirrors the implementation of the pages context menu I used as a referenced.
I do believe this should be `constexpr const char* asdf = "asdf"` though, else the compiler warns, rightfully so, because I could just do `asdf[0] = 'x'` afterwards without the additional `const`.

Reply:
Oh yeah sure that const makes absolute sense!
Using the old implementation as a reference is a double-edged sword. While it is good to look at it, there is no promise that it is really that good. We have some serious mistakes in the codebase that are only slowly removed over time.

Reply:
Writing `constexpr auto* asdf = "asdf";` is also valid.